src/symbol/mod.rs

```diff
@@ -1250,6 +1254,92 @@ impl<'t> TryFromCtx<'t, SymbolKind> for BlockSymbol<'t> {
     }
 }
 
+/// Thunk adjustor
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct ThunkAdjustor<'t> {
+    delta: u16,
+    target: RawString<'t>,
+}
+
+/// A thunk kind
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ThunkKind<'t> {
+    /// Standard thunk
+    NoType,
+    /// "this" adjustor thunk with delta and target
+    Adjustor(ThunkAdjustor<'t>),
+    /// Virtual call thunk with table entry
+    VCall(u16),
+    /// pcode thunk
+    PCode,
+    /// thunk which loads the address to jump to via unknown means...
+    Load,
+    /// Unknown
+    Unknown,
+}
+
+/// A thunk symbol.
+///
+/// Symbol kind `S_THUNK32`, or `S_THUNK32_ST`.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct ThunkSymbol<'t> {
+    /// The parent scope that this thunk is nested in.
+    pub parent: Option<SymbolIndex>,
+    /// The end symbol of this thunk.
+    pub end: SymbolIndex,
+    /// The next symbol.
+    pub next: Option<SymbolIndex>,
+    /// Code offset of the start of this label.
+    pub offset: PdbInternalSectionOffset,
+    /// The length of the thunk.
+    pub len: u16,
+    /// The kind of the thunk.
+    pub kind: ThunkKind<'t>,
+    /// The thunk name.
+    pub name: RawString<'t>,
+}
+
+impl<'t> TryFromCtx<'t, SymbolKind> for ThunkSymbol<'t> {
+    type Error = Error;
+    type Size = usize;
+
+    fn try_from_ctx(this: &'t [u8], kind: SymbolKind) -> Result<(Self, Self::Size)> {
+        let mut buf = ParseBuffer::from(this);
+
+        let parent = parse_optional_index(&mut buf)?;
+        let end = buf.parse()?;
+        let next = parse_optional_index(&mut buf)?;
+        let offset = buf.parse()?;
+        let len = buf.parse()?;
+        let ord = buf.parse::<u8>()?;
+        let name = parse_symbol_name(&mut buf, kind)?;
+
+        let kind = match ord {
+            0 => ThunkKind::NoType,
+            1 => ThunkKind::Adjustor(ThunkAdjustor {
+                delta: buf.parse::<u16>()?,
+                target: buf.parse_cstring()?,
+            }),
+            2 => ThunkKind::VCall(buf.parse::<u16>()?),
+            3 => ThunkKind::PCode,
+            4 => ThunkKind::Load,
+            _ => ThunkKind::Unknown,
```

Comment:
Should we error out here instead? Currently, such cases are handled inconsistently in the crate. In the TPI this would be handled with `unreachable!()`, which is definitely worse.

Reply:
Maybe just use a Unknown(u8) instead of Unknown and let the user decides what he wants to do with it. Wdyt ?

Reply:
`Unknown(u8)` sounds good 👍 

At first, I thought that parsing of the symbol depends on the kind. Since the additional data is all after name, it doesn't really matter if we don't understand a certain kind. Sorry for this oversight.